src/test/netlink/test_ifinfomsg.rs

```diff
@@ -0,0 +1,217 @@
+        let seq = Some(0xfe182ab9); // Random number
+        let pid = None;
+        let payload = NlPayload::Payload(ifinfomsg);
+        Nlmsghdr::new(len, nl_type, flags, seq, pid, payload)
+    };
+
+    let mut buffer = Cursor::new(Vec::new());
+    nlmsg.to_bytes(&mut buffer).unwrap();
+    let buffer = buffer.into_inner();
+
+    let ret = unsafe {
+        libc::sendto(
+            fd,
+            buffer.as_ptr() as *const core::ffi::c_void,
+            buffer.len(),
+            0,
+            core::ptr::null(),
+            0,
+        )
+    };
+    if ret < 0 {
+        return Err(anyhow!("sento error"));
+    }
+
+    let mut buffer = vec![0; 8192];
+    let ret = unsafe {
+        libc::recvfrom(
+            fd,
+            buffer.as_mut_ptr() as *mut core::ffi::c_void,
+            buffer.len(),
+            0,
+            core::ptr::null_mut(),
+            core::ptr::null_mut(),
+        )
+    };
+    if ret < 0 {
+        return Err(anyhow!("recvfrom error"));
+    }
+    buffer.truncate(ret as usize);
+
+    let Ok(nlmsg) = Nlmsghdr::<Rtm, Ifinfomsg>::from_bytes(&mut Cursor::new(buffer.as_slice()))
+    else {
+        return Err(anyhow!("failed to deserialize the message"));
+    };
+    let Ok(_ifinfomsg) = nlmsg.get_payload() else {
+        return Err(anyhow!("failed to find the payload"));
+    };
+
+    Ok(())
+}
+
+fn test_shorter_than_nlmsghdr() -> anyhow::Result<()> {
+    let fd = unsafe {
+        libc::socket(
+            libc::AF_NETLINK,
+            libc::SOCK_RAW | libc::SOCK_NONBLOCK,
+            libc::NETLINK_ROUTE,
+        )
+    };
+
+    // buffer is 15 bytes long which is shorter than nlmsghdr which is 16 bytes long
+    // 18 is RTM_GETLINK
+    let buffer: Vec<u8> = vec![15, 0, 0, 0, 18, 0, 1, 3, 185, 42, 24, 254, 0, 0, 0];
+
+    let ret = unsafe {
+        libc::sendto(
+            fd,
+            buffer.as_ptr() as *const core::ffi::c_void,
+            buffer.len(),
+            0,
+            core::ptr::null(),
+            0,
+        )
+    };
+    if ret < 0 {
+        return Err(anyhow!("sento error"));
+    }
+
+    let mut buffer = vec![0; 8192];
+    let ret = unsafe {
+        libc::recvfrom(
+            fd,
+            buffer.as_mut_ptr() as *mut core::ffi::c_void,
+            buffer.len(),
+            0,
+            core::ptr::null_mut(),
+            core::ptr::null_mut(),
+        )
+    };
+    if ret < 0 {
+        return Ok(());
+    }
+    Err(anyhow!("expected recvfrom error"))
+}
+
+fn test_shorter_than_ifinfomsg() -> anyhow::Result<()> {
+    let fd = unsafe {
+        libc::socket(
+            libc::AF_NETLINK,
+            libc::SOCK_RAW | libc::SOCK_NONBLOCK,
+            libc::NETLINK_ROUTE,
+        )
+    };
+
+    // buffer is 17 bytes long which is shorter than nlmsghdr+ifinfomsg which is 32 bytes long
+    // 18 is RTM_GETLINK
+    let buffer: Vec<u8> = vec![17, 0, 0, 0, 18, 0, 1, 3, 185, 42, 24, 254, 0, 0, 0, 0, 0];
```

Comment:
Like above, can you build this as a `nlmsghdr`+`ifinfomsg` struct and then truncate?